### PR TITLE
public amlip_cpp API [14547]

### PR DESCRIPTION
**Pull request by jparisu**
After #38

### CODE REVIEW

File: amlip_cpp/include/amlip_cpp/types/id/AmlipIdDataType.hpp [hunk 1]
```diff
@@ -27,7 +27,7 @@
 #include <ostream>
 #include <string>
```
Comment (irenebm) [inline amlip_cpp/include/amlip_cpp/types/id/AmlipIdDataType.hpp:29]:
```suggestion
#include <iostream>

```

File: amlip_cpp/include/amlip_cpp/node/StatusNode.hpp [hunk 2]
```diff
@@ -49,13 +66,21 @@ class StatusNode : public ParentNode
...
      *
      * @pre There could only be called once per instance before calling \c stop_processing
      */
     void process_status_async(
             const std::function<void(const types::StatusDataType&)>& callback);
 
+    /**
+     * Execute in a new thread a passive listening in Status topic and execute the functor given with each message
+     *
+     * @pre There could only be called once per instance before calling \c stop_processing
```
Comment (juanlofer-eprosima) [inline amlip_cpp/include/amlip_cpp/node/StatusNode.hpp:79]:
```suggestion
     * @pre Should only be called once per instance before calling \c stop_processing
```

File: amlip_cpp/src/cpp/types/id/idl/id.idl [hunk 3]
```diff
@@ -0,0 +1,6 @@
+
+struct AmlipId
```
Comment (juanlofer-eprosima) [inline amlip_cpp/src/cpp/types/id/idl/id.idl:2]:
```suggestion
struct AmlipIdDataType
```

**Review by juanlofer-eprosima — CHANGES_REQUESTED** [review]
Uncrustify

**Pull request merged**